pkg/kclient/generators.go

```diff
@@ -0,0 +1,62 @@
+package kclient
+
+import (
+
+	// api resource types
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenerateContainer creates a container spec that can be used when creating a pod
+func GenerateContainer(name, image string, isPrivileged bool, command, args []string, envVars []corev1.EnvVar) *corev1.Container {
+	container := &corev1.Container{
+		Name:            name,
+		Image:           image,
+		ImagePullPolicy: corev1.PullAlways,
+
+		Command: command,
+		Args:    args,
+		Env:     envVars,
+	}
+
+	if isPrivileged {
+		container.SecurityContext = &corev1.SecurityContext{
+			Privileged: &isPrivileged,
+		}
+	}
+
+	return container
+}
+
+// GeneratePodTemplateSpec creates a pod template spec that can be used to create a deployment spec
+func GeneratePodTemplateSpec(podName, namespace, serviceAccountName string, labels map[string]string, containers []corev1.Container) *corev1.PodTemplateSpec {
+	podTemplateSpec := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: serviceAccountName,
+			Containers:         containers,
+		},
+	}
+
+	return podTemplateSpec
+}
+
+// GenerateDeploymentSpec creates a deployment spec
+func GenerateDeploymentSpec(podTemplateSpec corev1.PodTemplateSpec) *appsv1.DeploymentSpec {
+	replicas := int32(1)
```

Comment:
Can't convert it directly to a constant because it is expected to be a pointer. So the best I could do is have a `const deploymentReplicas = 1` and then do something like this in the function:
```
func GenerateDeploymentSpec(podTemplateSpec corev1.PodTemplateSpec) *appsv1.DeploymentSpec {
	rep := deploymentReplicas
	labels := podTemplateSpec.ObjectMeta.Labels
	deploymentSpec := &appsv1.DeploymentSpec{
		Replicas: &rep,
...
```

But then I realized Replicas is optional and defaults to `1` so I just removed it :)

Reference: https://github.com/kubernetes/kubernetes/blob/6ecdd22978783ea51efc34f56f13eb4d9baa446a/pkg/apis/apps/types.go#L283

Reply:
Makes sense